controllers/auth_config_controller.go

```diff
@@ -231,15 +231,15 @@ func (r *AuthConfigReconciler) translateAuthConfig(ctx context.Context, authConf
 			if identity.APIKey.AllNamespaces && r.ClusterWide() {
 				namespace = ""
 			}
-			translatedIdentity.APIKey = identity_evaluators.NewApiKeyIdentity(identity.Name, identity.APIKey.LabelSelectors, namespace, authCred, r.Client, ctxWithLogger)
+			translatedIdentity.APIKey = identity_evaluators.NewApiKeyIdentity(identity.Name, identity.APIKey.Selector.MatchLabels, namespace, authCred, r.Client, ctxWithLogger)
```

Comment:
I know this is just a draft, so leaving here a comment rather for planning our next move than anything else.

Despite the very welcome enhancement to the API, to this point, effectively nothing really changes behaviour-wise.

In the end, `MatchLabels` are still a type of `map[string]string`, with extra built-in intelligence to work as label selectors based on the equality operator. In `APIKey.loadSecrets()`, we cast the map into `MatchingLabels`:

https://github.com/Kuadrant/authorino/blob/c10843f5307a4aae635992837ed4f200dc6fce67/pkg/evaluators/identity/api_key.go#L51

...which implement `client.ListOptions interface{}` using the equality operator applied to each keys-value pair: 

https://github.com/kubernetes-sigs/controller-runtime/blob/f236f0345ad2933912ebf34bfcf0f93620769654/pkg/client/options.go#L411

https://github.com/kubernetes/apimachinery/blob/1d209c836ad2fed6332e0aa08aefc669a649ba8e/pkg/labels/selector.go#L942

<br/>

What I imagine for a possible improvement here would still be a `selector.internalSelector` struct, but perhaps parsing from a label selector string:

https://github.com/kubernetes/apimachinery/blob/1d209c836ad2fed6332e0aa08aefc669a649ba8e/pkg/labels/selector.go#L869

Similar to how we did for the Authorino AuthConfig and Secret selectors:

https://github.com/Kuadrant/authorino/blob/c10843f5307a4aae635992837ed4f200dc6fce67/controllers/label_selector.go#L39

I'm talking about our own implementation of the `client.ListOptions` interface possibly, with the label selectors parsed from a string capable of `client.ApplyToList()`.

This would allow using other operators besides equality. See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels.

Reply:
Done in https://github.com/Kuadrant/authorino/pull/341/commits/90c30b2af551ec763476846e6ab85bbd7391f53d